mypy/plugins/functools.py

```diff
@@ -0,0 +1,105 @@
+"""Plugin for supporting the functools standard library module."""
+from typing import Dict, NamedTuple, Optional
+
+import mypy.plugin
+from mypy.nodes import ARG_OPT, ARG_POS, ARG_STAR2, Argument, FuncItem, Var
+from mypy.plugins.common import add_method_to_class
+from mypy.types import AnyType, CallableType, get_proper_type, Type, TypeOfAny, UnboundType
+
+
+functools_total_ordering_makers = {
+    'functools.total_ordering',
+}
+
+_ORDERING_METHODS = {
+    '__lt__',
+    '__le__',
+    '__gt__',
+    '__ge__',
+}
+
+
+_MethodInfo = NamedTuple('_MethodInfo', [('is_static', bool), ('type', CallableType)])
+
+
+def functools_total_ordering_maker_callback(ctx: mypy.plugin.ClassDefContext,
+                                            auto_attribs_default: bool = False) -> None:
+    """Add dunder methods to classes decorated with functools.total_ordering."""
+    if ctx.api.options.python_version < (3, 2):
+        ctx.api.fail('"functools.total_ordering" is not supported in Python 2 or 3.1', ctx.reason)
```

Comment:
Or 3.0 I guess. It doesn't really matter since mypy doesn't support any of these early Python 3 versions, so this could just check for Python 2 too.